diff --git a/nodes/trellis2/model.py b/nodes/trellis2/model.py
index 06a3776..21de305 100644
--- a/nodes/trellis2/model.py
+++ b/nodes/trellis2/model.py
@@ -1541,13 +1541,13 @@ class SparseStructureFlowModel(nn.Module):
 
         if pe_mode == "ape":
             pos_embedder = AbsolutePositionEmbedder(model_channels, 3)
-            coords = torch.meshgrid(*[torch.arange(res, device=self.device) for res in [resolution] * 3], indexing='ij')
+            coords = torch.meshgrid(*[torch.arange(res, device=device) for res in [resolution] * 3], indexing='ij')
             coords = torch.stack(coords, dim=-1).reshape(-1, 3)
             pos_emb = pos_embedder(coords)
             self.register_buffer("pos_emb", pos_emb)
         elif pe_mode == "rope":
             pos_embedder = RotaryPositionEmbedder(self.model_channels // self.num_heads, 3)
-            coords = torch.meshgrid(*[torch.arange(res, device=self.device) for res in [resolution] * 3], indexing='ij')
+            coords = torch.meshgrid(*[torch.arange(res, device=device) for res in [resolution] * 3], indexing='ij')
             coords = torch.stack(coords, dim=-1).reshape(-1, 3)
             rope_phases = pos_embedder(coords)
             self.register_buffer("rope_phases", rope_phases)
